Source/Core/InputCommon/ControllerInterface/evdev/evdev.cpp

```diff
@@ -72,7 +72,7 @@ static void HotplugThreadFunc()
     FD_SET(monitor_fd, &fds);
     FD_SET(s_wakeup_eventfd, &fds);
 
-    int ret = select(monitor_fd + 1, &fds, nullptr, nullptr, nullptr);
+    int ret = select(std::max(monitor_fd, s_wakeup_eventfd) + 1, &fds, nullptr, nullptr, nullptr);
```

Comment:
Should the `continue`s below (starting after the !devnode continue) also drop the reference to the device with `udev_device_unref`? As far as I know, if `udev_monitor_receive_device` returns a non-null pointer, then it needs to be passed to the unref function.

Either way, LGTM.

Reply:
>On success, udev_monitor_receive_device() returns a pointer to a newly referenced device that was received via the monitor. The caller is responsible to drop this reference when done. On failure, NULL is returned.

Whoops, indeed! I've added a commit to take care of that.